### DISCUSSION TITLE
LoRa override sync-byte (sync-word)

### DISCUSSION THREAD

**Discussion by keever50** [zephyrproject-rtos/zephyr#78995] [Ideas]
**tldr**
The LoRa API does not support changing the LoRa sync-byte, which is required in some LoRa implementations.
Overriding using a function can solve this issue.

**Problem**
I have a big issue porting this project "Meshtastic" to Zephyr because the LoRa API does not allow changing the sync-byte (or called sync-word) to more than 2 hard-coded bytes, which are selected with a bool called PublicNetwork.
Meshtastic the protocol uses 0x2B as sync-word to not interfere with other protocols.
Is there a reason for this 2 restrictive hard-coded bytes?

**Dirty workaround**
I got around this problem by changing the hard-coded byte manually in the driver header, but this is far from ideal.

**Possible solutions**
The solution would be to allow the programmer to enter the sync word instead of a single bool. 
Or have a function that lets the programmer force override the sync-word without breaking the compatibility of current implementation.

**Snippets**
![image](https://github.com/user-attachments/assets/0d1b17f5-870b-4bdf-ab19-de5a5e5edee0)
Here i modified the private sync-word to 0x2B
![image](https://github.com/user-attachments/assets/0748af76-9aa3-46c2-8f59-6492cfe48c1e)
Here you see the bool switch that selects two hard-coded bytes. This is where the main issue lays.

**Comment by keever50** [comment]
Also for reference:
![image](https://github.com/user-attachments/assets/4acbc841-6f72-4fc0-bd96-61f0424a1525)
